trie/encoding.go

```diff
@@ -75,13 +75,16 @@ func keybytesToHex(str []byte) []byte {
 }
 
 // hexToKeybytes turns hex nibbles into key bytes.
-// This can only be used for keys of even length.
+// This is only useful for keys of even length.
 func hexToKeybytes(hex []byte) []byte {
 	if hasTerm(hex) {
 		hex = hex[:len(hex)-1]
 	}
 	if len(hex)&1 != 0 {
-		panic("can't convert hex key of odd length")
+		// Odd length. This can happen if someone iterates a Trie
```

Comment:
It's explained in the commit message:

> This also removes the the panic for odd length because it is too subtle
> to be useful. In a follow-up commit, using LeafKey in a core/state test
> revealed that it iterates tries rooted in an intermediate node. This can
> produce odd-length leaf keys and it's hard to debug. Removing the panic
> removes risk of a rare production crash later.